src/Shared/FileUtilitiesRegex.cs

```diff
@@ -19,13 +15,21 @@ namespace Microsoft.Build.Shared
     /// </summary>
     internal static class FileUtilitiesRegex
     {
-        // regular expression used to match file-specs beginning with "<drive letter>:" 
-        internal static readonly Regex DrivePattern = new Regex(@"^[A-Za-z]:", RegexOptions.Compiled);
+        // regular expression used to match file-specs comprising exactly "<drive letter>:" (with no trailing characters)
+        internal static readonly Regex DrivePattern = new Regex(@"^[A-Za-z]:$", RegexOptions.Compiled);
+
+        private static readonly string s_baseUncPattern = string.Format(
+            CultureInfo.InvariantCulture,
+            @"^[\{0}\{1}][\{0}\{1}][^\{0}\{1}]+[\{0}\{1}][^\{0}\{1}]+",
```

Comment:
Is pattern hasn't changed, right?

Reply:
UNC paths were sometimes checked in 'prefix' mode, sometimes in 'exact' mode.

This base pattern is the same. I create two Regex objects from it, and one has a `$` appended, for exact match.